File: max_length_diff/go/max_length_diff.go
package main

import (
	"fmt"
	"math"
)

func main() {
	//a1 := []string{}
	a1 := []string{"hoqq", "bbllkw", "oox", "ejjuyyy", "plmiis", "xxxzgpsssa", "xxwwkktt", "znnnnfqknaz", "qqquuhii", "dvvvwz"}
	a2 := []string{"cccooommaaqqoxii", "gggqaffhhh", "tttoowwwmmww"}

	if len(a1) == 0 || len(a2) == 0 {
		fmt.Println(-1)
		return
	}

	var diffs []float64
	for _, strA := range a1 {
		for _, strB := range a2 {
			diff := float64(len(strA) - len(strB))
			diffs = append(diffs, math.Abs(diff))
		}
	}
	max := diffs[0]
	for _, value := range diffs {
		if value > max {
			max = value
		}
	}
	println(int(max))
}

File: max_length_diff/elixir/max_length_diff.exs
defmodule MaxDiff do
  def mxdiflg(a1, a2) do
    cond do
      length(a1) === 0 || length(a2) === 0 ->
        -1
      true ->
        getResult(a1, a2)
    end
  end

  defp getResult(a1, a2) do
    lengths_a = a1 |> Enum.map(&String.length/1)
    lengths_b = a2 |> Enum.map(&String.length/1)

    lengths_a
    |> Enum.map(fn item_a ->
      lengths_b |> Enum.map(fn item_b -> abs(item_a - item_b) end) |> Enum.max()
    end)
    |> Enum.max()
  end
end

a1 = [
  "hoqq",
  "bbllkw",
  "oox",
  "ejjuyyy",
  "plmiis",
  "xxxzgpsssa",
  "xxwwkktt",
  "znnnnfqknaz",
  "qqquuhii",
  "dvvvwz"
]

a2 = ["cccooommaaqqoxii", "gggqaffhhh", "tttoowwwmmww"]

IO.inspect(MaxDiff.mxdiflg(a1, a2))
